realm/realm-library/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -112,431 +112,304 @@ public TableQuery not() {
+        rawDescriptor(descriptor);
         return this;
     }
 
-    public TableQuery lessThan(long[] columnKey, long[] tablePtrs, double value) {
-        nativeLess(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
+    public static String buildDistinctDescriptor(String[] fieldNames) {
+        StringBuilder descriptorBuilder = new StringBuilder("DISTINCT(");
 
-    public TableQuery lessThanOrEqual(long[] columnKey, long[] tablePtrs, double value) {
-        nativeLessEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
-    }
+        String distinctSeparator = "";
+        for (String fieldName : fieldNames) {
+            descriptorBuilder.append(distinctSeparator)
+                    .append(escapeFieldName(fieldName));
 
-    public TableQuery between(long[] columnKey, double value1, double value2) {
-        nativeBetween(nativePtr, columnKey, value1, value2);
-        queryValidated = false;
-        return this;
-    }
+            distinctSeparator = ", ";
+        }
 
-    // Query for boolean values.
+        descriptorBuilder.append(")");
 
-    public TableQuery equalTo(long[] columnKey, long[] tablePtrs, boolean value) {
-        nativeEqual(nativePtr, columnKey, tablePtrs, value);
-        queryValidated = false;
-        return this;
+        return descriptorBuilder.toString();
     }
 
-    // Queries for Date values.
-
-    private static final String DATE_NULL_ERROR_MESSAGE = "Date value in query criteria must not be null.";
-
-    public TableQuery equalTo(long[] columnKey, long[] tablePtrs, @Nullable Date value) {
-        if (value == null) {
-            nativeIsNull(nativePtr, columnKey, tablePtrs);
-        } else {
-            nativeEqualTimestamp(nativePtr, columnKey, tablePtrs, value.getTime());
-        }
-        queryValidated = false;
+    public TableQuery distinct(String[] fieldNames) {
+        String descriptor = buildDistinctDescriptor(fieldNames);
+        rawDescriptor(descriptor);
         return this;
     }
 
-    public TableQuery notEqualTo(long[] columnKey, long[] tablePtrs, Date value) {
-        //noinspection ConstantConditions
-        if (value == null) { throw new IllegalArgumentException(DATE_NULL_ERROR_MESSAGE); }
-        nativeNotEqualTimestamp(nativePtr, columnKey, tablePtrs, value.getTime());
-        queryValidated = false;
+    public TableQuery limit(long limit) {
+        rawDescriptor("LIMIT(" + limit + ")");
         return this;
     }
 
-    public TableQuery greaterThan(long[] columnKey, long[] tablePtrs, Date value) {
-        //noinspection ConstantConditions
-        if (value == null) { throw new IllegalArgumentException(DATE_NULL_ERROR_MESSAGE); }
-        nativeGreaterTimestamp(nativePtr, columnKey, tablePtrs, value.getTime());
+    public TableQuery isEmpty(String fieldName) {
+        rawPredicateWithPointers(escapeFieldName(fieldName) + ".@count = 0");
```

Comment:
Does this also work with Strings? If I remember correctly the old `isEmpty()` also accepted Strings?

Reply:
Yes, it does!